### ISSUE TITLE
Version  0.6.6 bluetooth track not working

### ISSUE THREAD

**Issue by alesoft73** [custom-components/sensor.mitemp_bt#69]
After update , the bluetooth tracking don't work in version 0.6.6 (the sensor xiami they work).

Return 0.6.5  RESTART.
For return work the bluetooth tracking on 0.6.5.  i have unistall,  reboot ha , reinstall 0.6.5 .
After work perfect.
Nothing error.

For information i have hassio on VM

**Comment by Magalex2x14** [comment]
In version 0.6.6, compared with 0.6.5, there are no changes in the reception of data that can affect the operation of third-party components. I suspect this is a coincidence. As [described in our FAQ](https://github.com/custom-components/sensor.mitemp_bt/blob/master/faq.md#conflicts-with-other-components-using-the-same-bt-interface), the reliable operation of this component jointly with other bt-components (provided that one bt-interface is used) is practically impossible, it all depends on which component will gain access to the interface faster during HA launch. This is almost a lottery. The only thing you can experiment within this situation is the periodicity of the components (the `period` option for our component and a similar option for a third-party component, if any).

**Comment by alesoft73** [comment]
The problem is that the 0.6.6 component is interfering with the HA system's bluetooth scan.
Blocking the detection of other bluetooth devices.
With 0.6.5 it works without problems.
I haven't changed anything in my configuration ...
Thanks for your work and support.

**Comment by Magalex2x14** [comment]
Perhaps I did not put it very clearly. Do you read the FAQ on the link that I gave above?
The bottom line is that this component has always interfered with others using the same bluetooth interface. You are lucky that it worked for you earlier.
In version 0.6.6 there are no changes in operation with the bluetooth subsystem.

**Comment by andras-tim** [comment]
> Perhaps I did not put it very clearly. Do you read the FAQ on the link that I gave above?
> The bottom line is that this component has always interfered with others using the same bluetooth interface. You are lucky that it worked for you earlier.
> In version 0.6.6 there are no changes in operation with the bluetooth subsystem.

@Magalex2x14, I have a slightly connected question (sorry for the off):
I use your plugin for a **LYWSDCGQ** and a **LYWSD02** 2 weeks ago without any problems - I love it, thank you! :slightly_smiling_face: 
Unfortunately, the clock of my **LYWSD02** is always hurrying, therefore I decided to reset the clock every midnight with https://github.com/h4/lywsd02 library. But now I read your previous comment and I'm afraid of that. Can I run my script once a day? Should I restart something after the run? Or should I count with exceptions, freezes, ...? - You know, I would like to avoid buying a second dongle for this small task :confused:

**Comment by Magalex2x14** [comment]
@andras-tim I think you should try it. The problem with my component and trackers is that they both work continuously and periodically change the bt-interface mode, interfering with each other's work. From what I managed to find out about this problem, I assume that starting a third-party component once a day may go unnoticed for my component - I assume that the maximum that you can lose is data for one period (60 seconds by default) in whole or in part.
In general, try it. And in any case, please keep me informed - create a separate issue when you do it.

**Comment by alesoft73** [comment]
> 
> 
> Perhaps I did not put it very clearly. Do you read the FAQ on the link that I gave above?
> The bottom line is that this component has always interfered with others using the same bluetooth interface. You are lucky that it worked for you earlier.
> In version 0.6.6 there are no changes in operation with the bluetooth subsystem.

Are you telling me that the component does not work in Has bluetooth for normal tracking?
Because I repeat, sorry, but with the 0.6.5. it is working too
bluetooth tracking of home assistant.
I don't think it's a coincidence. Or am I wrong ?

**Comment by Magalex2x14** [comment]
> Are you telling me that the component does not work in Home Assistant bluetooth for normal tracking?

If you mean the simultaneous work of my component and `bluetooth_tracker`, then yes, unfortunately, in all cases I knew of, normal simultaneous work of these components could not be achieved. And so it was from the very beginning. And this comes from the principles of operation of both components. Take a look at this question: #23.

Perhaps there is some dependence on the model of the bt adapter used (during the collection of information about the problem, I read that there are chips that allow simultaneous scanning and connection). But on my two systems, I was not able to achieve normal operation - either my component or tracker worked, and never together.

> I don't think it's a coincidence. Or am I wrong?

We are all human and prone to mistakes. But, I repeat, version 0.6.6 does not contain any changes in working with the bluetooth subsystem. And I really see no reason why updating a component has such an effect on your system. I assume that it depends on the specific moment the components are launched, and I have no way to influence this.

If you are technically savvy enough and ready to tackle this problem, then to shed light on your situation, you need to do some research:
- it’s worth starting with your configuration - post here the config lines related to the tracker and my component;
- carefully check the logs for any errors when working with version 0.6.6;
- roll back to version 0.6.5 and write down the debug and `btmon` log [as described in our FAQ](https://github.com/custom-components/sensor.mitemp_bt/blob/master/faq.md#debug), in the DEBUG section;
- install 0.6.6 and write down the same debug and `btmon` logs as in the previous paragraph, and put both here;

Only after I have enough detailed information I can make some conclusions about your situation. Until then, I suggest you stay on version 0.6.5 if you do not experience any problems with it.

I should also note that the forum has a lot of complaints about the work of the bluetooth tracker. Many people solve these problems using an external device (ESP32, for example, with ESPHome firmware) or switch to tracking via WiFi, which works more stably and has a larger coverage area.

**Comment by alesoft73** [comment]
With 0.6.7beta bluetooth track
Work perfect....
Many thanks
